docs/core-api/BLOCK.md

```diff
@@ -73,19 +74,18 @@ An optional object which may have the following keys:
 | ---- | ---- | ------- | ----------- |
 | format | `String` | `'dag-pb'` | The codec to use to create the CID |
 | mhtype | `String` | `sha2-256` | The hashing algorithm to use to create the CID |
-| mhlen | `Number` | | |
+| mhlen | `Number` | `undefined` | The hash length (only relevant for `go-ipfs`) |
```

Comment:
https://github.com/ipfs/js-ipfs/blob/master/packages/ipfs-core-types/src/block/index.ts does not include this option. However, the `go-ipfs` CLI seems to support this option and found this mentions in this repository:

https://github.com/ipfs/js-ipfs/blob/548cfb215e8afa8dca64d3c38fa13feecbe577e2/packages/ipfs-cli/src/commands/block/put.js#L34-L37

https://github.com/ipfs/js-ipfs/blob/548cfb215e8afa8dca64d3c38fa13feecbe577e2/packages/ipfs-http-server/src/api/resources/block.js#L111

- Is this option relevant? Should it be removed?
- Is it supported by `js-ipfs` or only `go-ipfs`?

Reply:
It's not supported by `js-ipfs` and I'm not clear on what it even does for `go-ipfs`.  It [defaults to `-1`](https://docs.ipfs.io/reference/http/api/#api-v0-block-put) which means.. 🤷 

If you could find out the what and the why maybe we could even remove it entirely.

Reply:
Well, it seems to be literally the [multihash length](https://github.com/ipfs/go-ipfs/search?q=mhlen), but doesn't seem to be very useful looking at the very limited search results. I tried figuring out if this was useful when opening this PR, but couldn't find anything that explained why it may be needed.

Reply:
I had to refactor relevant go-ipfs code around this recently `mhlen`  gets passed deep into go-multihash code responsible for digert (`Sum`) calculation: https://github.com/multiformats/go-multihash/blob/75ae3688857d036ea15947be9df68d3172b19470/sum.go


tldr: 
- `-1` is the implicit default which means "set  length based on hashing function output"
  - same applies to any value `<=0`
- setting custom `mhlen`  allows for truncating  longer digests produces by alternative hash functions 
- setting it to longer than the default output of specific hash function will produce error